synapse/cortex.py

```diff
@@ -6179,6 +6199,557 @@ async def iterTagPropRows(self, layriden, tag, prop, form=None, stortype=None, s
+
+    def _delVaultByIden(self, iden):
+        idenb = s_common.uhex(iden)
+        vault = self._getVaultByIdenb(idenb)
+
+        self.slab.delete(vault.get('type').encode(), idenb, db=self.vaultsbytypedb)
+        self.slab.delete(vault.get('name').encode(), db=self.vaultsbynamedb)
+        return self.slab.delete(idenb, db=self.vaultsdb)
+
+    async def setVaultDefault(self, vtype, scope, user=None):
+        '''
+        Set a default scope for a vault type.
+
+        Permissions: vaults.set
+
+        Args:
+            vtype (str): Name of the type to set.
+            scope (str|None): The scope to set as default or None for no default.
+            user (Optional[HiveUser]): User trying to set the default value.
+
+        Raises:
+            synapse.exc.BadArg:
+                - `vtype` does not exist.
+                - `default` is not one of: None, 'user', 'role', 'global'.
+                - A `vtype` vault with `default` scope does not exist.
+
+            synapse.exc.AuthDeny: `user` does not have permission to set vault type values.
+
+        Returns: (boolean) True if default was set, false otherwise.
+        '''
+        if scope not in (None, 'user', 'role', 'global'):
+            raise s_exc.BadArg(mesg=f'Invalid scope value: {scope})')
+
+        if user is None:
+            user = self.auth.rootuser
+
+        user.confirm(('vaults', 'defaults'))
+
+        if scope is not None:
+            # Scan for a vault of this specified type with the specified scope
+            for vault in self._getVaultsByType(vtype):
+                if vault.get('scope') == scope:
+                    break
+
+            else:
+                raise s_exc.BadArg(mesg=f'Vault type with {scope} scope does not exist')
+
+        return await self._push('cortex:vault:default', vtype, scope)
+
+    @s_nexus.Pusher.onPushAuto('cortex:vault:default')
+    async def _setVaultDefault(self, vtype, scope):
+        if scope is None:
+            return self.slab.delete(vtype.encode(), db=self.vaultdefaultsdb)
+        else:
+            return self.slab.put(vtype.encode(), s_msgpack.en(scope), db=self.vaultdefaultsdb)
+
+    def getVaultByIden(self, iden, user=None):
+        '''
+        Get a vault by iden.
+
+        Permissions: `user` must have at least `PERM_EDIT` for the requested vault.
+
+        Args:
+            name (str): Iden of the vault to get.
+            user (Optional[HiveUser]): User trying to get the vault.
+
+        Raises:
+            synapse.exc.NoSuchIden: Vault with `name` (name or iden) is not found.
+
+        Returns: vault or None if user doesn't have permission to requested vault.
+        '''
+        if user is None:
+            user = self.auth.rootuser
+
+        vault = self._getVaultByIden(iden, throw=True)
+
+        if self._hasEasyPerm(vault, user, s_cell.PERM_EDIT):
+            return vault
+
+        if self._hasEasyPerm(vault, user, s_cell.PERM_READ):
+            vault.pop('data')
```

Comment:
Why is this `data` key being popped here?

Reply:
Because if you only have read access to the vault, you aren't allowed to see the contents (the data).